src/Pages/VideoListing/VideoListing.jsx

```diff
@@ -1,31 +1,110 @@
+import { useLocation, useSearchParams } from "react-router-dom";
+import { useEffect } from "react";
 import { VerticalCard } from "../../Components/Cards/VerticalCard";
 import FloatingButton from "../../Components/UI/Button/FloatingButton";
-import { useVideo } from "../../Context";
+import { useFilter, useVideo } from "../../Context";
+import { finalFilteredData } from "../../Utils/finalFilteredData";
 import "./VideoListing.css";
 
 export const VideoListing = () => {
   const {
     videoState: { videos, categories },
   } = useVideo();
+  const { filterState, filterDispatch } = useFilter();
+  const { byCategory, bySearch, byLatest } = filterState;
+  const { search } = useLocation();
+  const urlParam = new URLSearchParams(search);
+  const searchQuery = urlParam.get("query");
+
+  const onCategoryClickHandler = (category) => {
```

Comment:
Just a suggestion:  you could omit "on" just make it categoryClickHandler and for other functions as well?